src/mercury_engine_data_structures/formats/lua.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import typing
+
+import construct
+
+from mercury_engine_data_structures.common_types import StrId
+from mercury_engine_data_structures.formats import BaseResource
+from mercury_engine_data_structures.game_check import Game
+
+if typing.TYPE_CHECKING:
+    import typing_extensions
+
+    from mercury_engine_data_structures.file_tree_editor import FileTreeEditor
+
+LUA = construct.Struct(
+    lua_text = StrId,
+)
+
+
+class Lua(BaseResource):
+    @classmethod
+    def parse(cls, data: bytes, target_game: Game, editor: FileTreeEditor | None = None) -> typing_extensions.Self:
+        raise ValueError("Lua files cannot be parsed because it requires decompilation")
+
+    def build(self) -> bytes:
+        if self.target_game == Game.SAMUS_RETURNS:
+            import randovania_lupa.lua51_sr as lupa
+        else: # self.target_game == Game.DREAD
```

Comment:
would it make sense to have a
```py
elif game == game.dread
 ...
else
  raise exception("unknown game")
```
? Not sure whether the other mercury engine games (i.e. lords of shadow) would need this

Reply:
Done. For other mercury engine games it will probably explode at 300 different places anyways. Though it would be interesting if they used the same version for any of the other games.